### PR TITLE
Updated algorithm names

### LINKED ISSUE #5498: CMM Algorithm names
Since we have not deleted algorithms 2 & 3, I think we need to rename Algorithm 1 and just use 'MSH Algorithm' instead of Algorithm 4.    People will wonder what are algorithms 2 & 3!

### PR DESCRIPTION
**Pull request by jmcameron**
This PR updated the CMM/AMC algorithm names from "Algorithm 1" and "Algorithm 4 (MSH)" to "Default Algorithm" and "MSH Algorithm" respectively.   It also cleans out some left-over code related to algorithms 2 and 3, including some end-to-end tests that might have been broken before this change.  All of the tests work on my system--except the end-to-end test which fails on my system exactly like the master branch has been doing for some time.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5498

**Testing**
- Try the Articles in Stock page and view the CMM calculation dialog (double-click on CMM on a row)
- Try changing the CMM algorithm in the Stock Settings page and verify that it all looks good.
- Go back to the Articles in Stock page and verify that the algorithm changed worked.
- Please verify that the French translations are okay.

### CODE REVIEW

File: test/end-to-end/stock/stock.setting.js [hunk 1]
```diff
@@ -21,16 +21,12 @@ function StockSettingTests() {
...
     await page.setRadio('yes', 'enable_auto_stock_accounting');
     await page.setRadio('yes', 'enable_supplier_credit');
-    await element(by.id('algo_msh')).click();
 
-    const checkSelectAlgo1 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(0);
-    const checkSelectAlgo2 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(1);
-    const checkSelectAlgo3 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(2);
-    const checkSelectAlgoMsh = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(3);
+    await element(by.id('algo_msh')).click();
+    const checkSelectAlgoDef = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(0);
```
Comment (jniles) [inline test/end-to-end/stock/stock.setting.js:26]: Awesome 👍

**Review by jniles — APPROVED** [review]
Looks fine to me.  I'll let @mbayopanda give his opinion.

File: client/src/i18n/fr/stock.json [hunk 2]
```diff
@@ -140,10 +140,10 @@
     "MONTHLY_CONSUM"  : "Consommation Mensuelle Moyenne",
     "MONTHLY_CONSUMPTION" : {
       "AVERAGE_MONTHLY_CONSUMPTION"  : "Consommation Mensuelle Moyenne",
-      "ALGO_1"          : "Algorithme 1",
-      "ALGO_1_COMMENT"  : "La consommation moyenne mensuel est obtenue en divisant la quantité consommée durant la période définie par le nombre de jour de stock pour ladite période, et en multipliant le résultat obtenu par 30.5",
-      "ALGO_4"          : "Algorithme 4 (MSH)",
-      "ALGO_4_COMMENT"  : "La consommation moyenne est égale à : La quantité consommée durant la période / (Nombre de mois à considérer - (Le nombre de jour de rupture de stock / 30.5)). L'algorithme MSH est recommandé par l'organisation Management Sciences for Health (https://www.msh.org).",
+      "ALGORITHM_DEFAULT"          : "Algorithme par défaut ",
```
Comment (mbayopanda) [inline client/src/i18n/fr/stock.json:143]: 👍🏽

File: test/end-to-end/stock/stock.setting.js [hunk 3]
```diff
@@ -46,16 +42,12 @@ function StockSettingTests() {
...
     await page.setRadio('no', 'enable_supplier_credit');
-    await element(by.id('algo3')).click();
 
-    const checkSelectAlgo1 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(0);
-    const checkSelectAlgo2 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(1);
-    const checkSelectAlgo3 = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(2);
-    const checkSelectAlgoMsh = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(3);
+    await element(by.id('algo_def')).click();
+    const checkSelectAlgoDef = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(0);
+    const checkSelectAlgoMsh = element.all(by.model('StockSettingsCtrl.settings.average_consumption_algo')).get(1);
```
Comment (mbayopanda) [inline test/end-to-end/stock/stock.setting.js:48]: 👍🏽

**Review by mbayopanda — COMMENTED** [review]
The code LGTM 

I have noticed that default Algorithm has nothing as value : 
image

**Comment by jmcameron** [comment]
> I have noticed that default Algorithm has nothing as value :

Good catch!   I'll fix that.

**Comment by jmcameron** [comment]
> 
> 
> > I have noticed that default Algorithm has nothing as value :
> 
> Good catch! I'll fix that.

This turned out not to be a bug.  The code that evaluates the default algorithm value is in SQL so the database needs to be refreshed then the value will show up.  Or try switching to a production database.   Don't forget to run

CALL zRecomputeStockMovementStatus();

in MySQL to have data for the CMM calculations.   Not sure if you need to run that with bhima_test, but you do need to run it (I think) with production databases (after installing the migration file SQL fixes).   Perhaps @jniles could comment on this.

@mbayopanda Please try this again.  If it is okay, I think it is ready to go.

**Comment by mbayopanda** [comment]
Here is what I got without running `CALL zRecomputeStockMovementStatus();`
image

After running the procedure, I still have this blank. I am running this on IMCK database and I got this : 
image

All CMM calculation is disappeared

**Comment by jmcameron** [comment]
> All CMM calculation is disappeared

Not sure what is going on here.   I think we may need to pull in @jniles !

**Comment by mbayopanda** [comment]
@jmcameron I recreate the production database for a second test, and I still have nothing as the default algorithm 
image

The problem with CMM data that disappear in the registry is not related to this PR I think since I cannot reproduce that after reinstalling the production database

**Comment by jniles** [comment]
Everything works fine on my end:

image

@mbayopanda this may be a cache related issue.  Please make sure you've done the following:
  1. Build the database.
  2. Run the migration script.
  3. Rebuild the server
  4. Cleared your browser cache.

If it survives that, I don't know what to think.  It shouldn't have anything to do with StockMovementStatus, only the `GetAMC()` procedure.

My current hypothesis is that you are caching something like the default algorithm and this is causing issues for you.

**Pull request merged**